frontend/src/helpers/get-fingerprint.js

```diff
@@ -0,0 +1,39 @@
+import Fingerprint2 from 'fingerprintjs2';
+import UAParser from 'ua-parser-js';
+
+export const getFingerprint = () => {
+  return new Promise((resolve, reject) => {
+    async function getHash() {
```

Comment:
FP shouldn't be calculated on each invocation, it should persist on local storage or at least session storage

Reply:
the fingerprint is now calculated only if it is not present in the localStorage